### PR TITLE
AB#87732 Summary of Delivery - Check answers

### PR DESCRIPTION
**Pull request by Carq**
image

image

### CODE REVIEW

File: src/HE.Investment.AHP.WWW/Models/Delivery/Factories/DeliveryPhaseSummaryViewModelFactory.cs [hunk 1]
```diff
@@ -86,35 +86,34 @@ public class DeliveryPhaseSummaryViewModelFactory : IDeliveryPhaseSummaryViewMod
...
-        DeliveryPhaseDetails deliveryPhase)
+    private static SectionSummaryViewModel CreateMilestonesSummary(DeliveryPhaseDetails deliveryPhase)
     {
+        var summary = deliveryPhase.SummaryOfDelivery;
         var items = new List<SectionSummaryItemModel>
         {
             new(
                 "Grant apportioned to this phase",
-                "TODO".ToOneElementList(),
+                $"\u00a3{summary?.GrantApportioned.ToPoundsPencesString() ?? "-"}".ToOneElementList(),
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW/Models/Delivery/Factories/DeliveryPhaseSummaryViewModelFactory.cs:96]: Please create formatter extension method to wrap money formatting logic.

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain.Tests/Delivery/Entities/TestDataBuilders/DeliveryPhaseEntityBuilder.cs [hunk 2]
```diff
@@ -68,9 +68,9 @@ public DeliveryPhaseEntityBuilder WithUnregisteredBody()
         return this;
     }
 
-    public DeliveryPhaseEntityBuilder WithDeliveryPhaseMilestones(DeliveryPhaseMilestones milestones)
+    public DeliveryPhaseEntityBuilder WithDeliveryPhaseMilestones(DeliveryPhaseMilestones? milestones = null)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain.Tests/Delivery/Entities/TestDataBuilders/DeliveryPhaseEntityBuilder.cs:71]: new DeliveryPhaseMilestonesBuilder().Build() is default Builder behaviour 
there is no need to to add it here.
Now someone using WithDeliveryPhaseMilestones(null) may think that we can set DeliveryMilestonesTo null.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain.Tests/Delivery/Entities/TestDataBuilders/DeliveryPhaseEntityBuilder.cs (hunk 2 above)
Comment (Carq) [reply src/HE.Investment.AHP.Domain.Tests/Delivery/Entities/TestDataBuilders/DeliveryPhaseEntityBuilder.cs:71]: I need to tell explicitly that DelivePhase has Milestones, if someone wants to have null DeliveryMiless then they can use `WithoutDeliveryPhaseMilestones`

File: src/HE.Investment.AHP.WWW/Models/Delivery/Factories/DeliveryPhaseSummaryViewModelFactory.cs (hunk 1 above)
Comment (Carq) [reply src/HE.Investment.AHP.WWW/Models/Delivery/Factories/DeliveryPhaseSummaryViewModelFactory.cs:96]: JJ is doing that.

**Pull request merged**